### PR TITLE
fix: Load multiple models with the same project name

### PR DESCRIPTION
**Pull request by MoritzWeber0**
Resolves https://github.com/DSD-DBS/capella-dockerimages/issues/173

This json...

```json
[
  {
    "url": "https://github.com/DSD-DBS/coffee-machine",
    "revision": "main",
    "depth": 0,
    "entrypoint": "coffee-machine-demo.aird"
  },
  {
    "url": "https://github.com/DSD-DBS/coffee-machine",
    "revision": "main",
    "depth": 0,
    "entrypoint": "coffee-machine-demo.aird"
  },
  {
    "url": "https://github.com/DSD-DBS/coffee-machine",
    "revision": "diagram-cache-workflow",
    "depth": 0,
    "entrypoint": "coffee-machine-demo.aird"
  }
]
```

... produces this output: 
image

In general: 
- If there we have duplicates, add the sanitised revision to the project name.
- If there are still duplicates, count the projects and add a number as suffix.
- If we have even more duplicates, they are ignored.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -336,25 +336,15 @@ run-capella/readonly: capella/readonly
...
 		-p $(RDP_PORT):3389 \
 		-e RMT_PASSWORD=$(RMT_PASSWORD) \
-		-e GIT_URL=$(GIT_REPO_URL) \
-		-e GIT_ENTRYPOINT=$(GIT_REPO_ENTRYPOINT) \
-		-e GIT_REVISION=$(GIT_REPO_BRANCH) \
-		-e GIT_DEPTH=$(GIT_REPO_DEPTH) \
-		-e GIT_USERNAME="$(GIT_USERNAME)" \
-		-e GIT_PASSWORD="$(GIT_PASSWORD)" \
-		--entrypoint bash \
+		-e GIT_REPOS_JSON="$$(cat local/readonly.json)" \
```
Comment (dominik003) [inline Makefile:347]: Shouldn't we make the path here configurable? So use something like GIT_REPOS_JSON_PATH with the default value being `local/readonly.json`?

File: ease/Dockerfile [hunk 2]
```diff
@@ -110,7 +110,7 @@ WORKDIR /home/techuser
 
 COPY git_askpass.py /etc/git_askpass.py
 ENV GIT_ASKPASS=/etc/git_askpass.py
-RUN chmod +x /etc/git_askpass.py
+RUN chmod +rx /etc/git_askpass.py
```
Comment (dominik003) [inline ease/Dockerfile:113]: Why do we need read permissions here?

File: readonly/load_models.py [hunk 3]
```diff
@@ -11,34 +11,56 @@
     The acronym EASE stands for "Eclipse Advanced Scripting Environment".
     Further information: https://www.eclipse.org/ease/
 """
+import collections
```
Comment (dominik003) [inline readonly/load_models.py:14]: Can be removed as it is not used

File: readonly/load_models.py [hunk 4]
```diff
@@ -48,8 +70,8 @@ def fetch_projects_from_environment() -> list[dict[str, str]]:
             {
                 "url": git_repo_url,
                 "revision": git_repo_revision,
-                "depth": os.getenv("GIT_DEPTH"),
-                "entrypoint": os.getenv("GIT_ENTRYPOINT"),
+                "depth": int(os.getenv("GIT_DEPTH", "0")),
```
Comment (dominik003) [inline readonly/load_models.py:73]:
```suggestion
                "depth": int(os.getenv("GIT_DEPTH", 0)),
```

File: readonly/load_models.py [hunk 5]
```diff
@@ -86,59 +108,136 @@ def clone_git_model(project: dict[str, str]) -> None:
...
+        for duplicated_project in duplicated_projects:
+            append_revision_to_project_name(duplicated_project)
+
+    # Check again for duplicated projects (same project name and same revision)
+    duplicated_projects_grouped = group_projects_by_name(projects)
+    for duplicated_projects in duplicated_projects_grouped:
+        counter = 0
+        for duplicated_project in duplicated_projects:
+            add_suffix_to_project_name(duplicated_project, str(counter))
+            counter += 1
```
Comment (dominik003) [inline readonly/load_models.py:183]:
```suggestion
    for counter, duplicated_projects in enumerate(duplicated_projects_grouped):
        for duplicated_project in duplicated_projects:
            add_suffix_to_project_name(duplicated_project, str(counter))
```

File: readonly/load_models.py [hunk 6]
```diff
@@ -86,59 +108,136 @@ def clone_git_model(project: dict[str, str]) -> None:
...
+) -> list[list[_ProjectDict]]:
+    grouped_projects: dict[str, list[_ProjectDict]] = {}
+
+    for project in projects:
+        project_name = project["name"]
+        if project_name in grouped_projects:
+            grouped_projects[project_name].append(project)
+            continue
+
+        grouped_projects[project_name] = [project]
```
Comment (dominik003) [inline readonly/load_models.py:197]: Here, we could also simply use an `else` instead of the continue

**Review by dominik003 — COMMENTED** [review]
LGTM! Just a few comments that could be adjusted before merging.

**Review by dominik003 — APPROVED** [review]

File: readonly/load_models.py (hunk 4 above)
Comment (MoritzWeber0) [reply readonly/load_models.py:73]: `key, default and the result are str` (see docstring of getenv)

File: readonly/load_models.py (hunk 5 above)
Comment (MoritzWeber0) [reply readonly/load_models.py:183]: Nice one :)

Comment (MoritzWeber0) [reply readonly/load_models.py:183]: I just had to move the counter to the second loop.

**Pull request merged**